rbc/heavydb/table_function_manager.py

```diff
@@ -63,109 +67,58 @@ def set_output_array_values_total_number(self, index: int,
 
 
 error_msg = 'TableFunctionManager is only available in HeavyDB 5.9 or newer (got %s)'
-i8p = ir.IntType(8).as_pointer()
-i32 = ir.IntType(32)
-i64 = ir.IntType(64)
-
 
-@extending.intrinsic
-def heavydb_udtfmanager_error_message_(typingctx, mgr, msg):
-    sig = types.int32(mgr, msg)
 
+@extending.overload_method(TableFunctionManagerNumbaType, 'error_message')
+def heavydb_udtfmanager_error_message(mgr, msg):
     target_info = TargetInfo()
     if target_info.software[1][:3] < (5, 9, 0):
         raise UnsupportedError(error_msg % (".".join(map(str, target_info.software[1]))))
 
     if not isinstance(msg, types.StringLiteral):
         raise RequireLiteralValue(f"expected StringLiteral but got {type(msg).__name__}")
 
-    def codegen(context, builder, signature, args):
-        mgr_ptr = args[0]
-
-        mgr_i8ptr = builder.bitcast(mgr_ptr, i8p)
-
-        msg_bytes = msg.literal_value.encode('utf-8')
-        msg_const = make_bytearray(msg_bytes + b'\0')
-        msg_global_var = global_constant(
-            builder.module,
-            "table_function_manager_error_message",
-            msg_const)
-        msg_ptr = builder.bitcast(msg_global_var, i8p)
-
-        fnty = ir.FunctionType(i32, [i8p, i8p])
-        fn = cgutils.get_or_insert_function(
-            builder.module, fnty, "TableFunctionManager_error_message")
+    defn = 'int32_t TableFunctionManager_error_message(int8_t*, int8_t*)|CPU'
```

Comment:
Should we use `|CPU` suffix elsewhere as well?

Reply:
Yes! fixed